test/DisputeManager.test.ts

```diff
@@ -0,0 +1,207 @@
+        const deployment = await deployContracts(root, root);
+        disputeManager = deployment.disputeManager;
+        token = deployment.token;
+        staking = deployment.staking;
+        indexerRegistry = deployment.indexerRegistry;
+        eraManager = deployment.eraManager;
+        rewardsDistributor = deployment.rewardsDistributer;
+        rewardsStaking = deployment.rewardsStaking;
+        rewardsHelper = deployment.rewardsHelper;
+        await eraManager.updateEraPeriod(time.duration.days(5).toString());
+        await token.connect(fisherman).increaseAllowance(disputeManager.address, etherParse("1000"));
+        await registerIndexer(token, indexerRegistry, staking, root, indexer, '2000');
+    });
+
+    it('set MinimumDeposit should work', async () => {
+        expect(await disputeManager.minimumDeposit()).to.equal(etherParse('1000'));
+        await disputeManager.setMinimumDeposit(etherParse('10'));
+        expect(await disputeManager.minimumDeposit()).to.equal(etherParse('10'));
+    });
+
+    it('createDispute should work', async () => {
+        await token.connect(root).transfer(fisherman.address, etherParse("1000"));
+        let tx = await disputeManager.connect(fisherman).createDispute(indexer.address, DEPLOYMENT_ID, etherParse('1000'), 0);
+        let receipt = await tx.wait();
+        let evt = receipt.events.find(
+            (log) => log.topics[0] === utils.id('DisputeOpen(uint256,address,address,uint8)')
+        );
+        let event = disputeManager.interface.decodeEventLog(
+            disputeManager.interface.getEvent('DisputeOpen'),
+            evt.data
+        );
+        expect(event.fisherman).to.eql(fisherman.address);
+        expect(event.indexer).to.eql(indexer.address);
+        expect(event._type).to.eql(0);
+
+
+        expect(await disputeManager.nextDisputeId()).to.equal(2);
+        expect(await disputeManager.disputeIdByIndexer(indexer.address,0)).to.equal(1);
+
+        let dispute = await disputeManager.disputes(1);
+        expect(dispute.disputeId).to.equal(1);
+        expect(dispute.indexer).to.equal(indexer.address);
+        expect(dispute.fisherman).to.equal(fisherman.address);
+        expect(dispute.depositAmount).to.equal(etherParse('1000'));
+        expect(dispute.deploymentId).to.equal(DEPLOYMENT_ID);
+        expect(dispute.dtype).to.equal(0);
+        expect(dispute.state).to.equal(0);
+
+        expect(await token.balanceOf(fisherman.address)).equal(0);
+    });
+
+    it('createDispute not reach MinimumDeposit should fail', async () => {
+        await token.connect(root).transfer(fisherman.address, etherParse("1000"));
+        await expect(
+            disputeManager.connect(fisherman).createDispute(indexer.address, DEPLOYMENT_ID, etherParse('10'), 0)
+        ).to.be.revertedWith('Not meet the minimum deposit');
+    });
+
+    it('createDispute on an indexer over 20 times should fail', async () => {
```

Comment:
Can't remember why we need `require(disputeIdByIndexer[_indexer].length <= 20, 'reach dispute limit');`